### PR TITLE
feat: upgrade to prettier 3 and prettier-plugin-svelte 3

### LINKED ISSUE #10751: `npm run lint` causes a warning for every file: `[warn] Ignored unknown option { pluginSearchDirs: ["."] }.` due to Prettier
### Describe the bug

Running `npm run lint` causes this warning for every file: `[warn] Ignored unknown option { pluginSearchDirs: ["."] }.` 

Creating noisy output.

### Reproduction

1. Using SvelteKit's default `.prettierrc`:
```
{
	"useTabs": true,
	"singleQuote": true,
	"trailingComma": "none",
	"printWidth": 100,
	"plugins": ["prettier-plugin-svelte"],
	"pluginSearchDirs": ["."],
	"overrides": [{ "files": "*.svelte", "options": { "parser": "svelte" } }]
}
```

2. And `"lint": "prettier --plugin-search-dir . --check . && eslint .",`
3. Run `npm run format` (You can remove eslint, it's not related to that.)

Gives output of:

```
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
[warn] Ignored unknown option { pluginSearchDirs: ["."] }.
```

### Logs

_No response_

### System Info

```Shell
Binaries:
    Node: 20.4.0 - /opt/homebrew/bin/node
    Yarn: 1.22.4 - /usr/local/bin/yarn
    npm: 9.7.2 - /opt/homebrew/bin/npm
    pnpm: 8.6.1 - /opt/homebrew/bin/pnpm
  npmPackages:
    @sveltejs/adapter-cloudflare: ^2.3.3 => 2.3.3 
    @sveltejs/kit: ^1.25.0 => 1.25.0 
    svelte: ^4.2.0 => 4.2.0 
    vite: ^4.4.9 => 4.4.9 

// package.json related
    "prettier": "^3.0.3",
    "prettier-plugin-svelte": "^3.0.3",
    "prettier-plugin-tailwindcss": "^0.5.4",
    "eslint-config-prettier": "^9.0.0",
```

### Severity

annoyance

### Additional Information

Removing `"plugins": ["prettier-plugin-svelte"],` fixes it.

### PR DESCRIPTION
**Pull request by benmccann**
This will give users the new versions via `create-svelte`. The rest of the change is internal

closes #10751
closes https://github.com/sveltejs/kit/issues/10716

### CODE REVIEW

**Comment by astral4** [comment]
Shouldn't the template .prettierrc file [here](https://github.com/sveltejs/kit/blob/b251b235b321b9c75236429cc3a1ac2415475775/packages/create-svelte/shared/%2Bprettier/.prettierrc) also be changed? [`pluginSearchDirs`](https://prettier.io/blog/2023/07/05/3.0.0.html#plugin-search-feature-has-been-removed-14759httpsgithubcomprettierprettierpull14759-by-fiskerhttpsgithubcomfisker) has been completely removed from the API in Prettier 3.

**Comment by benmccann** [comment]
Okay. Seems this is blocked by https://github.com/prettier/prettier/issues/15079. There's a PR for it, so hopefully we don't have to wait too long: https://github.com/prettier/prettier/pull/15155

**Comment by benmccann** [comment]
https://github.com/prettier/prettier/pull/15433 was merged with the fix. It's not been released yet though. It looks like it will be included in prettier 3.1

**Comment by polyzen** [comment]
Prettier 3.1 has been released with the fix 🚀

File: packages/adapter-auto/package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
 		"adapters.js"
 	],
 	"scripts": {
-		"lint": "prettier --check . --config ../../.prettierrc --ignore-path .gitignore",
```
Comment (dummdidumm) [inline packages/adapter-auto/package.json:28]: I saw these were removed - could someone clarify why? Or why they were needed before? (probably related to the major version bump?)

**Review by dummdidumm — APPROVED** [review]

**Comment by hyunbinseo** [comment]
Finally! A migration guide would be nice for existing projects.

```diff
# jsconfig.json, tsconfig.json
+ "moduleResolution": "bundler"
```

```diff
# package.json
- "lint": "prettier --plugin-search-dir . --check . && eslint .",
- "format": "prettier --plugin-search-dir . --write ."
+ "lint": "prettier --check . && eslint .",
+ "format": "prettier --write ."
```

```diff
# package.json
- "prettier": "^2.8.8",
- "prettier-plugin-svelte": "^2.10.1",
+ "prettier": "^3.1.0",
+ "prettier-plugin-svelte": "^3.1.0",
```

```diff
# .prettierrc
- "pluginSearchDirs": ["."],
```

File: packages/create-svelte/shared/+eslint+prettier/package.json [hunk 2]
```diff
@@ -3,7 +3,7 @@
 		"eslint-config-prettier": "^8.5.0"
```
Comment (hyunbinseo) [inline packages/create-svelte/shared/+eslint+prettier/package.json:3]: Can't `eslint-config-prettier` be bumped to v9? It does not depend on Prettier v3, but it is the only outdated package as of now. [Changelog](https://github.com/prettier/eslint-config-prettier/blob/main/CHANGELOG.md)

Comment (benmccann) [reply packages/create-svelte/shared/+eslint+prettier/package.json:3]: yes. thanks for the suggestion. PR here: https://github.com/sveltejs/kit/pull/11037

**Comment by hyunbinseo** [comment]
The comment can work as a guide, but it should be added to the documentation.

Similar case: migration to `eslint-plugin-svelte` [docs](https://svelte.dev/docs/v4-migration-guide#new-eslint-package) links to a [GitHub comment](https://github.com/sveltejs/kit/issues/10242#issuecomment-1610798405).

---

I do wish there was an official guide to bumping packages' major versions.

If the user started with `create-svelte@3.0.0`, these packages are outdated.

| Package                                | Current | Latest | Migration                         |
| -------------------------------------- | ------- | ------ | --------------------------------- |
| @typescript-eslint/eslint-plugin (dev) | 5.62.0  | 6.11.0 | None                              |
| @typescript-eslint/parser (dev)        | 5.62.0  | 6.11.0 | None                              |
| eslint-config-prettier (dev)           | 8.10.0  | 9.0.0  | None - PR in process              |
| prettier (dev)                         | 2.8.8   | 3.1.0  | pacakge.json scripts, .prettierrc |
| prettier-plugin-svelte (dev)           | 2.10.1  | 3.1.0  | pacakge.json scripts, .prettierrc |
| svelte (dev)                           | 3.59.2  | 4.2.3  | Reference Svelte docs             |
| typescript (dev)                       | 4.9.5   | 5.2.2  | Mention moduleResolution: bundler |
| eslint-plugin-svelte3                  | 4.0.0   | -      | Should be uninstalled             |
| eslint-plugin-svelte                   | -       | 2.35.0 | Should be installed, [migration]  |

[migration]: https://github.com/sveltejs/kit/issues/10242#issuecomment-1610798405

This will allows users to update their dependencies with confidence.

**Comment by dummdidumm** [comment]
The place to add that documentation would be the `prettier-plugin-svelte` repository. It already contains a small section at the bottom of the readme. Happy to accept a PR with more migration instructions.

**Pull request merged**